### PR TITLE
Collect jdbc wallet

### PR DESCRIPTION
**Pull request by jshum2479**
The PR collects the wallet of the data source if the connection property is pointing to a directory or file in the archive during discovery

### CODE REVIEW

**Review by robertpatrick — APPROVED** [review]

**Review by CarolynRountree — APPROVED** [review]

**Review by rakillen — APPROVED** [review]

**Comment by ddsharpe** [comment]
<02-04-2023 00:20:04> <INFO> Executing shell command: target/weblogic-deploy/bin/archiveHelper.sh add databaseWallet  -source src/test/resources/cwallet.sso -wallet_name acmeWallet  -archive_file target/test-output/test35DiscoverDomainWithJDBCWallet/gen-wallet-archive.zip

<02-04-2023 00:20:04> <INFO> Test log: target/test-output/test35DiscoverDomainWithJDBCWallet

<02-04-2023 00:20:04> <INFO> Executing shell command: target/weblogic-deploy/bin/createDomain.sh -oracle_home /u01/oracle -domain_home ./target/domains/domain2-discover35 -model_file src/test/resources/simple-topology-jdbcwallet.yaml -archive_file target/test-output/test35DiscoverDomainWithJDBCWallet/gen-wallet-archive.zip

<02-04-2023 00:20:15> <SEVERE> ========== FAILED test [Test 35: test create and discover domain with jdbc wallet] method=test35DiscoverDomainWithJDBCWallet: 0 min, 11 sec  ==========

File: core/src/main/python/wlsdeploy/tool/util/archive_helper.py [hunk 1]
```diff
@@ -401,6 +401,20 @@ def extract_database_wallet(self, wallet_name=WLSDeployArchive.DEFAULT_RCU_WALLE
         self.__logger.exiting(class_name=self.__class_name, method_name=_method_name, result=resulting_wallet_path)
         return resulting_wallet_path
 
+    def extract_all_database_wallets(self):
+        archive_entries = self.get_archive_entries()
+        wallet_names = sets.Set()
+        for entry in archive_entries:
+            if entry.startswith(WLSDeployArchive.ARCHIVE_DB_WALLETS_DIR):
+                if os.path.isdir(entry):
+                    name = wallet_names.add(os.path.basename(entry))
```
Comment (ddsharpe) [inline core/src/main/python/wlsdeploy/tool/util/archive_helper.py:410]: Set().add() does not return anything, so name will be void for this code path.  I don't think that is what you intended.

**Review by ddsharpe — CHANGES_REQUESTED** [review]

File: core/src/main/python/wlsdeploy/tool/util/archive_helper.py (hunk 1 above)
Comment (jshum2479) [reply core/src/main/python/wlsdeploy/tool/util/archive_helper.py]: That's unintentional, fixed

File: core/src/main/python/wlsdeploy/tool/discover/common_resources_discoverer.py [hunk 2]
```diff
@@ -112,11 +114,126 @@ def get_datasources(self):
...
+
+        onprem_wallet_dir_is_not_flat = self._is_on_prem_wallet_dir_flat(onprem_wallet_parent_path, property_value)
+        archive_file = self._model_context.get_archive_file()
+
+        # keep track of the wallet name and from where it is collected
+        #  { <onprem_wallet_parentpath> : { 'wallet_name': <name>, 'path_into_archive': <wallet path into archive>,
+        #      },
+        #     .... }
+        #
+        if not onprem_wallet_parent_path in collected_wallet_dictionary:
```
Comment (ddsharpe) [inline core/src/main/python/wlsdeploy/tool/discover/common_resources_discoverer.py:175]: How does the not operator work on a path?  Did you mean to use the `not in` operator? (from Sonar code smells)

Comment (jshum2479) [reply core/src/main/python/wlsdeploy/tool/discover/common_resources_discoverer.py:175]: it's not a path, it is checking item in a dictionary

**Review by robertpatrick — APPROVED** [review]

**Pull request merged**